hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/ContainerEndpoint.java

```diff
@@ -498,46 +499,105 @@ private List<ContainerBlockMetadata> getBlocks(
+      // Send back an empty response
+      return Response.status(Response.Status.NOT_ACCEPTABLE).build();
+    }
+
     List<ContainerDiscrepancyInfo> containerDiscrepancyInfoList =
         new ArrayList<>();
     try {
       Map<Long, ContainerMetadata> omContainers =
           reconContainerMetadataManager.getContainers(-1, -1);
       List<Long> scmNonDeletedContainers =
           containerManager.getContainers().stream()
-              .filter(containerInfo -> !(containerInfo.getState() ==
-                  HddsProtos.LifeCycleState.DELETED))
-              .map(containerInfo -> containerInfo.getContainerID()).collect(
-                  Collectors.toList());
+              .filter(containerInfo -> containerInfo.getState() !=
+                  HddsProtos.LifeCycleState.DELETED)
+              .map(containerInfo -> containerInfo.getContainerID())
+              .collect(Collectors.toList());
 
-      // Filter list of container Ids which are present in OM but not in SCM.
+      // Filter list of container IDs which are present in OM but not in SCM.
       List<Map.Entry<Long, ContainerMetadata>> notSCMContainers =
-          omContainers.entrySet().stream().filter(containerMetadataEntry ->
-                  !(scmNonDeletedContainers.contains(
-                      containerMetadataEntry.getKey())))
-              .collect(
-                  Collectors.toList());
+          omContainers.entrySet().stream()
+              .filter(
+                  containerMetadataEntry -> !scmNonDeletedContainers.contains(
+                      containerMetadataEntry.getKey()))
+              .collect(Collectors.toList());
+
+      // Filter containers based on pagination parameters
+      if (prevKey > 0) {
+        int index = 0;
+        while (index < notSCMContainers.size() &&
+            notSCMContainers.get(index).getKey() <= prevKey) {
+          index++;
+        }
+        if (index < notSCMContainers.size()) {
+          notSCMContainers = notSCMContainers.subList(index,
+              Math.min(index + limit, notSCMContainers.size()));
+        } else {
+          notSCMContainers = Collections.emptyList();
+        }
+      } else {
+        notSCMContainers = notSCMContainers.subList(0,
+            Math.min(limit, notSCMContainers.size()));
+      }
 
       notSCMContainers.forEach(nonSCMContainer -> {
         ContainerDiscrepancyInfo containerDiscrepancyInfo =
             new ContainerDiscrepancyInfo();
         containerDiscrepancyInfo.setContainerID(nonSCMContainer.getKey());
         containerDiscrepancyInfo.setNumberOfKeys(
             nonSCMContainer.getValue().getNumberOfKeys());
-        containerDiscrepancyInfo.setPipelines(nonSCMContainer.getValue()
-            .getPipelines());
+        containerDiscrepancyInfo.setPipelines(
+            nonSCMContainer.getValue().getPipelines());
         containerDiscrepancyInfo.setExistsAt("OM");
         containerDiscrepancyInfoList.add(containerDiscrepancyInfo);
```

Comment:
@ArafatKhan2198 Thanks for working on this patch, however I think we need to return prevKey in output.